contributors/devel/sig-testing/verify-tests.md

```diff
@@ -0,0 +1,78 @@
+# Verification Tests
+
+**Table of Contents**
+
+- [Verification Tests](#verification-tests)
+  - [Overview](#overview)
+  - [`verify-govet-leveee`](#verify-govet-levee)
+
+## Overview
+
+PLACEHOLDER
+
+Verification tests for Kubernetes provide a mechanism to verify contributions for adherence to project conventions
+and best practices, and to validate generated build artifacts for soundness.
+
+Verification tests are found in vestigial shell scripts at `hack/verify-*.sh`.
+Most have been replaced by a corresponding `make` command.
+
+Most verification tests are self-explanatory.
+`verify-golint` and `verify-gofmt`, for instance, fail when a contribution does not adhere to lint and formatting conventions.
+More complex verification tests are described below.
+
+### `verify-govet-levee`
+
+Verification in `verify-govet-levee.sh` uses taint propagation analysis
+to defend against accidental logging of credentials.
+Struct fields which may contain credentials are annotated as such using the `datapolicy` field tag.
+Field tagging was introduced by [KEP-1753](https://github.com/kubernetes/enhancements/issues/1753), and analysis was introduced by [KEP-1993](https://github.com/kubernetes/enhancements/issues/1933).
+
+Taint propagation analysis defends against both direct and indirect logging of credentials.
+For consider the following hypothetical snippet
+
+```golang
+// kubernetes/cmd/kubelet/app/server.go
+
+// kubeConfigSpec struct holds info required to build a KubeConfig object
+type kubeConfigSpec struct {
+	CACert         *x509.Certificate
+	APIServer      string
+	ClientName     string
+	TokenAuth      *tokenAuth      `datapolicy:"token"`
+	ClientCertAuth *clientCertAuth `datapolicy:"security-key"`
+}
+
+func MyDangerousFunction(spec kubeConfigSpec) error {
+	if spec.CACert == nil {
+		err := fmt.Errorf("kubeConfigSpec missing expected CACert, got %#v", spec)
+		klog.Error(err)
+		return err
+	}
+	
+	if err := DoSomethingElse(spec); err != nil {
+		klog.Error(err)
+	}
+
+	return nil
+}
+```
+
+In the above, when `spec.CACert == nil`, we log the spec.
+However, we know from the datapolicy field tags that the spec could contain one or more credentials and should not be logged.
+Here, we would expect analysis to fail, and the log call should be adjusted to extract exactly what information is relevant from the `spec`.
```

Comment:
Nit: I think it would help to be more verbose with "we would expect analysis to fail". From the context it is not entirely clear that this means "the analysis will detect that this is an error and cause the verification test to fail".

Reply:
Rephrased.  It does read a bit better, thanks.